src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -214,40 +217,111 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
+                }
+                else
+                {
+                    _installedPackages = new PackageReference[0];
+                }
+
+                _lasTimeAssetsModified = fileInfo.LastWriteTimeUtc;
+            } 
 
-            return Task.FromResult<IEnumerable<PackageReference>>(installedPackages);
+            return _installedPackages;
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec, PackageSpec assetsPackageSpec, IList<LockFileTarget> targets = null)
         {
             var frameworkSorter = new NuGetFrameworkSorter();
 
+            PackageSpec combinedPackageSpec = new PackageSpec();
+
             return packageSpec
-                .TargetFrameworks
-                .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName))
-                .GroupBy(p => p.PackageIdentity)
-                .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
-                .ToArray();
+               .TargetFrameworks
+               .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName, assetsPackageSpec, targets))
+               .GroupBy(p => p.PackageIdentity)
+               .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
+               .ToArray();
         }
 
-        private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework)
+        private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework, PackageSpec assetsPackageSpec, IList<LockFileTarget> targets)
         {
+            TargetFrameworkInformation packageSpec = default;
+            foreach (var packageSpecFramwork in assetsPackageSpec.TargetFrameworks)
+            {
+                if (packageSpecFramwork.FrameworkName == targetFramework)
+                {
+                    packageSpec = packageSpecFramwork;
+                    break;
+                }
+            }
+
             return libraries
                 .Where(l => l.LibraryRange.TypeConstraint == LibraryDependencyTarget.Package)
-                .Select(l => new BuildIntegratedPackageReference(l, targetFramework));
+                .Select(l => new BuildIntegratedPackageReference(l, targetFramework, GetVersionFromPackageSpec(l, targetFramework, packageSpec, targets)));
+        }
+
+        private static PackageIdentity GetVersionFromPackageSpec(LibraryDependency l, NuGetFramework targetFramework, TargetFrameworkInformation packageSpec, IList<LockFileTarget> targets)
+        {
+            var installedVersion = GetInstalledVersion(l, targetFramework, packageSpec, targets);
+
+            if(installedVersion == null)
```

Comment:
format: Ex. if (installed

You might have some other formatting problems, make sure to format your code.